src/geanyobject.c

```diff
@@ -32,6 +32,10 @@
  * Emitted just after loading main keyfile settings.
  */
 
+/**
+ * @file geanyobject.h
+ * Definition of the global GeanyObject */
+
```

Comment:
Am 06.06.2016 um 03:27 schrieb Matthew Brush:
> In src/geanyobject.c
> <https://github.com/geany/geany/pull/1038#discussion_r65829764>:
>
>> @@ -32,6 +32,10 @@
>>   * Emitted just after loading main keyfile settings.
>>   */
>>
>> +/**
>> + * @file geanyobject.h
>> + * Definition of the global GeanyObject */
>> +
>
> No point in adding this if the stuff in this file is @gironly.
>
>

Corrected that.

>
>>
>> +#define GEANY_TYPE_DOCUMENT (document_get_type())
>> +GType document_get_type (void);
>
> Do these need to go in the headers if only used by g-ir-scanner?


Is also used inside Geany for g_signal_new() calls.

>> +
>> +static void        *copy_(void *src) { return src; }
>> +static void         free_(void *doc) { }
>> +
>> +/** Gets the GBOxed-derived type of GeanyDocument
>
> Should make this and the others @gironly or not doxygen-commented.


Why? I don't want GI-scanned API and actual API to diverge. The 
_get_type() functions can be exported without problems. _get_type() 
itself doesn't imply GBoxed or GObject or anything, nor does it imply 
any memory management. It just says there's a an associated GType to 
that type name.

I did change the comment that the GType is derived from GBoxed so that 
we can freely change these things to GObject later on.